AsyncDisplayKit/Private/ASDisplayNode+UIViewBridge.mm

```diff
@@ -224,47 +246,31 @@ - (void)setFrame:(CGRect)rect
     // Checking if the transform is identity is expensive, so disable when unnecessary. We have assertions on in Release, so DEBUG is the only way I know of.
     ASDisplayNodeAssert(CATransform3DIsIdentity(self.transform), @"-[ASDisplayNode setFrame:] - self.transform must be identity in order to set the frame property.  (From Apple's UIView documentation: If the transform property is not the identity transform, the value of this property is undefined and therefore should be ignored.)");
 #endif
+    _view.frame = rect;
+  } else if (!nodeLoaded || (isMainThread && !setFrameDirectly)) {
```

Comment:
The logic of this conditional is too complex for my liking, Particularly for critical code, And for having three different clauses. It is not immediately obvious if there is a situation where none of the clauses would run, Although I trusted that they do cover of the possibility space because of who wrote the code :)

Can you think of a way to modify it, perhaps with a higher-level check that forks based on any one of the three BOOLs?  Then you would have to large conditionals, and within each of those, at most two others (but presumably 2 and 1)